components/CreateCycle/CreateCycle.jsx

```diff
@@ -50,20 +53,22 @@ export default ({ navigation }) => {
 
   const createNewCycle = async (newCycle) => {
     const currentUser = firebase.auth().currentUser.uid;
-
     const cycleRef = firebase.firestore()
       .collection('users')
       .doc(currentUser)
       .collection('cycles');
-
-    const cycleDoc = await cycleRef.add(newCycle);
+    let cycleDoc;
+    if (isNewCycle) {
+      cycleDoc = await cycleRef.add(newCycle);
+    } else {
+      cycleDoc = await cycleRef.doc(cycleID).update(newCycle);
+    }
 
     dispatch(actions.cycles.addCycle({
       workouts: newCycle.workoutIds,
       name: newCycle.name,
       id: cycleDoc.id,
     }));
-    navigation.goBack();
```

Comment:
Is there a reason you decided to move this out of this function. This causes it to abruptly add the cycle in after like half a second if you create a new cycle